tools/java-source-utils/src/test/java/com/microsoft/android/JavadocXmlGeneratorTest.java

```diff
@@ -19,7 +19,12 @@
 public final class JavadocXmlGeneratorTest {
 	@Test(expected = FileNotFoundException.class)
 	public void init_invalidFileThrows() throws FileNotFoundException, ParserConfigurationException, TransformerException, UnsupportedEncodingException {
-		try (JavadocXmlGenerator g = new JavadocXmlGenerator("/this/file/does/not/exist")) {
+		String invalidFilePath = "/this/file/does/not/exist";
+		String osName = System.getProperty("os.name");
+		if (osName.startsWith("Windows")) {
+			invalidFilePath = "XYZABC:\\this\\file\\does\\not\\exist";
```

Comment:
On Windows, wouldn't we want/shouldn't `C:\\this\\file\\does\\not\\exist` fail?  The primary point isn't that the filename *itself* is invalid -- which would be the case on Windows with `XYZABC:\\…` -- but rather that a "valid" file path refers a non-existent filesystem entry.

(Plus, on Unix, I'm not sure it's really *possible* to have a path which *itself* is invalid.  `NUL` is "valid" {kinda, sorta} in that everything up to the NUL is handled, including newlines!)

Reply:
I believe `C:\\this\\file\\does\\not\\exist` would not throw a `FileNotFoundException` when attempting to create a `PrintStream`, as the [mkdirs](https://github.com/xamarin/java.interop/blob/e17bf093b5991a6a538870d4b4199f521e1eae03/tools/java-source-utils/src/main/java/com/microsoft/android/JavadocXmlGenerator.java#L45) call makes the path valid. We could change this to use a single non common drive name prefix that doesn't exist (e.g. `X` or something like that), or we could maybe just ignore this test on Windows.

Reply:
…so now I'm confused, which isn't great: `@Test(expected=FileNotFoundException.class)` should be asserting that `FileNotFoundException` is thrown.  The fact that this test is *passing* means that `FileNotFoundException` is thrown.

No [`java.io.File`](https://docs.oracle.com/javase/7/docs/api/java/io/File.html) APIs document throwing `FileNotFoundException`, meaning `parent.mkdirs()` isn't throwing.

This leaves the [`PrintStream(File)` constructor](https://docs.oracle.com/javase/7/docs/api/index.html?java/io/File.html), which *is* documented as throwing `FileNotFoundException`:

> If the given file object does not denote an existing, writable regular file and a new regular file of that name cannot be created, or if some other error occurs while opening or creating the file

So how's this test work in the first place?

*Permissions*, apparently; consider:

```java
import java.io.*;

class ps {
    public static void main (String[] args) throws Throwable {
        for (String a : args) {
            File f = new File(a);
            System.out.println(a + " exists? " + f.exists());
            File p = f.getParentFile();
            System.out.println(a + " parent? " + (p != null));
            if (p != null)
                System.out.println("mkdirs on parent? " + p.mkdirs());
            new PrintStream(a);
        }
    }
}
```

If I run this as:

    java ps /this/does/not/exist

I get:

```
/this/does/not/exist exists? false
/this/does/not/exist parent? true
mkdirs on parent? false
Exception in thread "main" java.io.FileNotFoundException: /this/does/not/exist (No such file or directory)
	at java.base/java.io.FileOutputStream.open0(Native Method)
…
```

meaning `File.mkdirs()` (1) doesn't throw, but (2) doesn't *do* anything, because a non-root user can't create the directory `/this`.

I imagine the same would be true on Windows.

/me tests…

So on Windows, a non-Admin user *can* create `C:\This`.  Which is why this test will "fail" on Windows, because `File.mkdirs()` *succeeds*.  Which is entirely unexpected (to me).

:thinking:

Reply:
This, on the other hand, fails (as expected):

```
> mkdir "\Program Files\a"
Access is denied.
```

So maybe on Windows we need to do:

```java
if (osName.startsWith("Windows")) {
    invalidFilePath = "C:\\Program Files\\does\\not\\exist";
}
```

Reply:
I think the Microsoft hosted machine pools we use have special permissions that allow the "agent" user to have more access and control. I'm trying to look for a restricted path but we may ultimately want to skip these two tests on Windows.

Reply:
There are no restrictions in writing to `Program Files` or `Windows` on these machines, rather than seeking out a restricted path I've decided to ignore these tests when running on CI on Windows.